scripts/socketserver/echo.py

```diff
@@ -26,8 +27,23 @@ def connection_handler(connection, path):
         log("{} closed connection".format(connection_id))
 
 if __name__ == "__main__":
-    log("Starting server at {}:{}".format(HOST, PORT))
-    server = websockets.serve(connection_handler, HOST, PORT)
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--port",
+        help="Port to listen on",
+        type=int,
+        default=4444)
+    parser.add_argument("--delay",
+        help="Time in seconds that a connection will be delayed before establishing it",
+        type=int,
+        default=0)
+    args = parser.parse_args()
+
+    def delaying_process_request(path, request_headers):
+        time.sleep(args.delay)
+        return None
+
+    log("Starting server at {}:{}".format(HOST, args.port))
```

Comment:
Why not make a `--host` option too?

Reply:
No need to make the script more flexible than necessary